### ISSUE TITLE
ktorrent fails to build

### ISSUE THREAD

**Issue by sjau** [NixOS/nixpkgs#42067]
## Issue description

Ktorrent fails to build

```
[build CXX] comphelper/source/misc/backupfilehelper.cxx                                                                                                                                                                          
[ 95%] Building CXX object ktorrent/CMakeFiles/ktorrent_app.dir/ktorrent_app_autogen/mocs_compilation.cpp.o                                                                                                                      
In file included from /tmp/nix-build-ktorrent-5.1.0.drv-0/ktorrent-5.1.0/build/ktorrent/ktorrent_app_autogen/2NRMJ5X7RK/moc_torrentcreatordlg.cpp:9:0,                                                                           
                 from /tmp/nix-build-ktorrent-5.1.0.drv-0/ktorrent-5.1.0/build/ktorrent/ktorrent_app_autogen/mocs_compilation.cpp:10:                                                                                            
/tmp/nix-build-ktorrent-5.1.0.drv-0/ktorrent-5.1.0/build/ktorrent/ktorrent_app_autogen/2NRMJ5X7RK/../../../../ktorrent/dialogs/torrentcreatordlg.h:86:16: error: field 'update_timer' has incomplete type 'QTimer'               
         QTimer update_timer;                                                                                                                                                                                                    
                ^~~~~~~~~~~~                                                                                                                                                                                                     
In file included from /nix/store/gp2pyyxphvybdgbwd238vs5q3b76z5a6-qtbase-5.10.1-dev/include/QtWidgets/qwidget.h:44:0,                                                                                                            
                 from /nix/store/gp2pyyxphvybdgbwd238vs5q3b76z5a6-qtbase-5.10.1-dev/include/QtWidgets/qdialog.h:44,                                                                                                              
                 from /nix/store/gp2pyyxphvybdgbwd238vs5q3b76z5a6-qtbase-5.10.1-dev/include/QtWidgets/QDialog:1,                                                                                                                 
                 from /tmp/nix-build-ktorrent-5.1.0.drv-0/ktorrent-5.1.0/build/ktorrent/ktorrent_app_autogen/2NRMJ5X7RK/../../../../ktorrent/dialogs/addpeersdlg.h:25,                                                           
                 from /tmp/nix-build-ktorrent-5.1.0.drv-0/ktorrent-5.1.0/build/ktorrent/ktorrent_app_autogen/2NRMJ5X7RK/moc_addpeersdlg.cpp:9,                                                                                   
                 from /tmp/nix-build-ktorrent-5.1.0.drv-0/ktorrent-5.1.0/build/ktorrent/ktorrent_app_autogen/mocs_compilation.cpp:2:                                                                                             
/nix/store/gp2pyyxphvybdgbwd238vs5q3b76z5a6-qtbase-5.10.1-dev/include/QtGui/qwindowdefs.h:76:7: note: forward declaration of 'class QTimer'                                                                                      
 class QTimer;                                                                                                                                                                                                                   
       ^~~~~~                                                                                                                                                                                                                    
 8:55.36 In file included from ../../dist/include/nsTObserverArray.h:11:0,                                                                                                                                                       
 8:55.36                  from /tmp/nix-build-palemoon-27.9.2.drv-0/palemoon-src/dom/base/nsINode.h:17,                                                                                                                          
 8:55.36                  from /tmp/nix-build-palemoon-27.9.2.drv-0/palemoon-src/dom/base/nsIContent.h:10,                                                                                                                       
 8:55.36                  from /tmp/nix-build-palemoon-27.9.2.drv-0/palemoon-src/layout/base/nsIPresShell.h:30,                                                                                                                  
 8:55.36                  from /tmp/nix-build-palemoon-27.9.2.drv-0/palemoon-src/layout/base/nsPresContext.h:16,                                                                                                                 
 8:55.36                  from /tmp/nix-build-palemoon-27.9.2.drv-0/palemoon-src/layout/base/../style/nsRuleNode.h:14,                                                                                                           
 8:55.36                  from /tmp/nix-build-palemoon-27.9.2.drv-0/palemoon-src/layout/base/../style/nsStyleContext.h:13,                                                                                                       
 8:55.36                  from /tmp/nix-build-palemoon-27.9.2.drv-0/palemoon-src/layout/base/../generic/nsIFrame.h:25,                                                                                                           
 8:55.36                  from /tmp/nix-build-palemoon-27.9.2.drv-0/palemoon-src/layout/base/nsGenConList.h:11,                                                                                                                  
 8:55.36                  from /tmp/nix-build-palemoon-27.9.2.drv-0/palemoon-src/layout/base/nsQuoteList.h:12,                                                                                                                   
 8:55.36                  from /tmp/nix-build-palemoon-27.9.2.drv-0/palemoon-src/layout/base/nsCSSFrameConstructor.h:18,                                                                                                         
 8:55.36                  from /tmp/nix-build-palemoon-27.9.2.drv-0/palemoon-src/layout/base/nsCSSFrameConstructor.cpp:12,                                                                                                       
 8:55.36                  from /tmp/nix-build-palemoon-27.9.2.drv-0/palemoon-src/obj-x86_64-unknown-linux-gnu/layout/base/Unified_cpp_layout_base1.cpp:11:                                                                       
 8:55.36 ../../dist/include/nsTArray.h: In static member function 'static void nsCSSRendering::PaintGradient(nsPresContext*, nsRenderingContext&, nsStyleGradient*, const nsRect&, const nsRect&, const nsRect&, const CSSIntRect
&, const nsSize&)':                                                                                                                                                                                                              
 8:55.36 Warning: -Wmaybe-uninitialized in /tmp/nix-build-palemoon-27.9.2.drv-0/palemoon-src/obj-x86_64-unknown-linux-gnu/dist/include/nsTArray.h: 'position' may be used uninitialized in this function                         
 8:55.36 ../../dist/include/nsTArray.h:486:5: warning: 'position' may be used uninitialized in this function [-Wmaybe-uninitialized]                                                                                             
 8:55.36      new (static_cast<void*>(aE)) E(mozilla::Forward<A>(aArg));                                                                                                                                                         
 8:55.36      ^                                                                                                                                                                                                                  
 8:55.37 In file included from /tmp/nix-build-palemoon-27.9.2.drv-0/palemoon-src/obj-x86_64-unknown-linux-gnu/layout/base/Unified_cpp_layout_base1.cpp:20:0:                                                                     
 8:55.37 /tmp/nix-build-palemoon-27.9.2.drv-0/palemoon-src/layout/base/nsCSSRendering.cpp:2405:12: note: 'position' was declared here                                                                                            
 8:55.37      double position;                                                                                                                                                                                                   
 8:55.37             ^                                                                                                                                                                                                           
[build CXX] comphelper/source/misc/base64.cxx                                                                                                                                                                                    
make[2]: *** [ktorrent/CMakeFiles/ktorrent_app.dir/build.make:706: ktorrent/CMakeFiles/ktorrent_app.dir/ktorrent_app_autogen/mocs_compilation.cpp.o] Error 1                                                                     
make[1]: *** [CMakeFiles/Makefile2:2011: ktorrent/CMakeFiles/ktorrent_app.dir/all] Error 2                                                                                                                                       
make: *** [Makefile:141: all] Error 2                                                                                                                                                                                            
builder for '/nix/store/mkfc07b8pldrjrxqrdy1v0h32chpv240-ktorrent-5.1.0.drv' failed with exit code 2                                                                                                                             
cannot build derivation '/nix/store/d9cmg64pf7i4cgnmy2ivbx30pf4y7f42-system-path.drv': 1 dependencies couldn't be built                                                                                                          
cannot build derivation '/nix/store/0mw4sg24rd2s1yizpvgyr4v17dgccj6n-nixos-system-subi-18.09pre142796.4b649a99d84.drv': 1 dependencies couldn't be built                                                                         
error: build of '/nix/store/0mw4sg24rd2s1yizpvgyr4v17dgccj6n-nixos-system-subi-18.09pre142796.4b649a99d84.drv' failed 
```

### Steps to reproduce



## Technical details

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the
results.

 - system: `"x86_64-linux"`                                                                                                                                                                                                      
 - host os: `Linux 4.15.12, NixOS, 18.09pre133640.ea145b68a01 (Jellyfish)`                                                                                                                                                       
 - multi-user?: `yes`                                                                                                                                                                                                            
 - sandbox: `no`                                                                                                                                                                                                                 
 - version: `nix-env (Nix) 2.0`                                                                                                                                                                                                  
 - channels(hyper): `"nixos-18.03pre123638.1258d9a4914"`                                                                                                                                                                         
 - channels(root): `"nixos-18.09pre142796.4b649a99d84"`                                                                                                                                                                          
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos/nixpkgs`

**Comment by Mic92** [comment]
will be fixed by https://github.com/NixOS/nixpkgs/pull/41944/

**Comment by pSub** [comment]
Seems to be fixed with #41944 and #42071.
